server/action_context.go

```diff
@@ -0,0 +1,14 @@
+package main
+
+// ActionContext passed from action buttons
+type ActionContext struct {
+	SubscriptionURL string `json:"subscription_url"`
+	Action          string `json:"action"`
+}
+
+// Action type for decoding action buttons
+type Action struct {
```

Comment:
I am a little confused, but why are we using custom types instead of the `model` variety?